static/js/publisher/release/components/releasesConfirm.js

```diff
@@ -56,26 +57,111 @@ class ReleasesConfirm extends Component {
   render() {
     const { isLoading, percentage } = this.state;
     const {
+      releases,
       pendingReleases,
       pendingCloses,
       isProgressiveReleaseEnabled
     } = this.props;
-    const releasesCount = Object.keys(pendingReleases).length;
     const closesCount = pendingCloses.length;
 
     const isPercentageValid = +percentage > 0 && +percentage <= 100;
 
+    // These are only used locally and therefore use revId-channel as a key, rather
+    // than the usual nested [revId][channel] format of pending releases
+    const progressiveUpdates = {};
+    const newReleases = {};
+    const newReleasesToProgress = {};
+
+    Object.keys(pendingReleases).forEach(revId => {
```

Comment:
This seems like a bit of data transformation. Could it be separated to outside function? Or a selector on the state maybe?